### PR TITLE
Add Accessibility Tests

### LINKED ISSUE #2200: [FEATURE] Accessibility checks to run as part of automated testing
### Description

Playwright has accessibility testing built in, but we need to config it https://playwright.dev/docs/accessibility-testing

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by dan-mba**
## Fixes Issue

Closes #2200 

## Changes proposed

Add accessibility tests using axe-core plugin for Playwright

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2379"></a>

### CODE REVIEW

**Comment by dan-mba** [comment]
Accessibility issues on Home Page:

1. Add lang attribute to html tag
2. GitHub icon link needs aria-label attribute
3. Increase contrast ratio for LinkFree version (current 5.78, required 7)
4. Increase contrast ratio for "Profile Views" purple box (current 6.28, required 7)
5. Increase contrast ratio for  "Users" purple box (current 2.98, required 7)
6. Increase contrast ratio for "Eddie Jaoude's" link (current 3.68, required 7)
7. Increase contrast ratio for "instructions" link (current 3.68, required 7)
8. Increase contrast ratio for help button (current 6.28, required 7)

**Comment by dan-mba** [comment]
I created a node script to test all the profile link icon colors for compatibility with AA contrast.
I found 19 of 35 icons fail.

The results & my test code are on GitHub in [this repo](https://github.com/dan-mba/color-check-file)
I will look into creating a better testing profile to use for accessibility tests that uses all of the icons.

**Comment by eddiejaoude** [comment]
@dan-mba I worry conflicts might occur, could we merge this in sooner as I think this can add value as we go along, let me know what you think?

**Comment by dan-mba** [comment]
> @dan-mba I worry conflicts might occur, could we merge this in sooner as I think this can add value as we go along, let me know what you think?

@eddiejaoude I was hoping to wait until the tests pass before adding them. I could put together a PR to fix all the remaining issues in a day.

**Comment by eddiejaoude** [comment]
> @eddiejaoude I was hoping to wait until the tests pass before adding them. I could put together a PR to fix all the remaining issues in a day

ok no worries, no rush/stress please, I really appreciate what you are doing - let me know if I can do anything to help

**Comment by dan-mba** [comment]
> ok no worries, no rush/stress please, I really appreciate what you are doing - let me know if I can do anything to help

I had considered handling all the issues myself to begin with, but wanted to offer people the chance for green squares. All of the changes are 1 line, so I thought people might be interested. I got some interest, but little follow through. As is typical of web dev in general, shiny features draw more interest over accessibility.

**Comment by eddiejaoude** [comment]
> wanted to offer people the chance for green squares

love that @dan-mba 👍 

> All of the changes are 1 line, so I thought people might be interested

Do you want me to do some shoutouts on Discord and Twitter to get more people involved?

**Comment by dan-mba** [comment]
> Do you want me to do some shoutouts on Discord and Twitter to get more people involved?

I think for this issue, it is best for me to finish it up so we can get this PR merged.

I will remember to get you involved if progress is lacking the next time I open an issue looking for contributors.

**Comment by dan-mba** [comment]
As soon as #2545 gets merged, this PR should be ready for review.

File: mongo-compose.yml
Comment (eddiejaoude) [inline mongo-compose.yml:7]: Is this for people who only want to run mongo on Docker? We can add this to the docs (I can help and do in another PR)

Comment (dan-mba) [reply mongo-compose.yml:7]: It is a way to run mongo so you can run playwright.
It is more of a stopgap measure.

There should be a way to direct playwright to use the existing server from the docker-compose, but I haven't had time to scour the docs for the correct cli option.

Comment (eddiejaoude) [reply mongo-compose.yml:7]: Sure, sounds good 👍  Thanks @dan-mba

Comment (dan-mba) [reply mongo-compose.yml:7]: I checked the playwright docs.
You can specify a config file with the `--config` option.

I tried this, but it doesn't work because the global-teardown  tries to cleanup the test users in mongo.

Comment (eddiejaoude) [reply mongo-compose.yml:7]: No worries, these improvements can come in a later PR, what there is now is great 👍 

I will just wait for Emma to review then we can merge

**Review by eddiejaoude — DISMISSED** [review]
Looks good to me 👍  But I am keen to get @EmmaDawsonDev thoughts also

**Review by EmmaDawsonDev — DISMISSED** [review]
I don't have much experience with Playwrite but it looks good as far as I can tell. 

Once the Christmas/New Year rush has settled down I can always help out if there are any failing tests. I can also do some more manual testing with keyboard and screen reader.

**Review by eddiejaoude — COMMENTED** [review]
Awesome 🔥

**Pull request merged**
